openhcl/hcl/src/ioctl.rs

```diff
@@ -1240,11 +1238,82 @@ impl MshvHvcall {
                 ));
             }
             Some(Vtl::Vtl1) => {
-                // TODO: allowed registers for VTL1
-                todo!();
+                todo!("TODO: allowed registers for VTL1");
             }
             Some(Vtl::Vtl0) => {
-                assert!(matches!(name, HvX64RegisterName::GuestOsId));
+                // Only VTL-private registers can go through this path.
+                // VTL-shared registers have to go through the kernel (either
+                // via the CPU context page or via the dedicated ioctl), as
+                // they may require special handling there.
+                //
+                // Register access should go through the register page if
+                // possible (as a performance optimization). In practice,
+                // registers that are normally available on the register page
+                // are handled here only when it is unavailable (e.g., running
+                // in WHP).
+                assert!(matches!(
```

Comment:
All non-shared registers are private. So I'd prefer to see the inverse check here (validate we don't have any shared registers), rather than have to update this every time we access another register.